lint.swift

```diff
@@ -0,0 +1,547 @@
+                !collection.compactMap({ OtherType($0) }).isEmpty()
+            else {
+                return
+            }
+            """,
+            """
+            guard
+                let collection = viewModel.myCollection(),
+                !collection.compactMap({ OtherType($0) }).isEmpty() else {
+                return
+            }
+            """
+        ],
+        includeFilters: [swiftFiles],
+        excludeFilters: [testFiles]
+    )
+
+    // MARK: NavigationControllerVariableNaming
+    try Lint.checkFileContents(
+        checkInfo: "NavigationControllerVariableNaming: Always name your navigation controller variables with the suffix `NavCtrl` or just `navCtrl`.",
+        regex: #"(var|let) +(nc|navigationcontroller|navc|ncontroller|navcontroller)[ :][^\n=]*=\i"#,
+        matchingExamples: ["let nc =", "var navigationController =", "let navc =", "let ncontroller =", "var nc: MyNavigationController =", "let navController: MyNavigationController<T> = "],
+        nonMatchingExamples: ["let navCtrl =", "let navCtrl: MyViewController =", "var myNavCtrl: MyViewController<T> ="],
+        includeFilters: [swiftFiles],
+        excludeFilters: [testFiles]
+    )
+
+    // MARK: NavigationControllerViewNaming
+    try Lint.checkFileContents(
+        checkInfo: "NavigationControllerViewNaming: Don't call your navigation controllers like view controllers, use the suffix `NavCtrl` or just `navCtrl`.",
+        regex: #"(var|let) +\w*(?<![Nn]avCtrl) *: *\w+NavigationController\?? *="#,
+        matchingExamples: ["let viewCtrl: UINavigationController? = activeViewController != nil ? self.activeViewController : self.portraitViewCtrl"],
+        nonMatchingExamples: ["let myNavCtrl: UINavigationController? = activeViewController != nil ? self.activeViewController : self.portraitViewCtrl"],
+        includeFilters: [swiftFiles],
+        excludeFilters: [testFiles]
+    )
+
+    // MARK: NilCoalescingOperator
+    try Lint.checkFileContents(
+        checkInfo: "NilCoalescingOperator: Prefer nil coalescing operator over `variable != nil ? variable! : alternative`.",
+        regex: #"(\w+)\s*!=\s*nil\s*\?\s*\1!\s*:\s*(.*)"#,
+        includeFilters: [swiftFiles],
+        excludeFilters: [testFiles],
+        autoCorrectReplacement: "$1 ?? $2",
+        autoCorrectExamples: [
+            [
+                "before": "    let message = errorMessage != nil ? errorMessage! : L10n.Global.Info.success\n",
+                "after": "    let message = errorMessage ?? L10n.Global.Info.success\n"
+            ],
+            [
+                "before": "param: callFunction(errorMessage != nil ? errorMessage! : L10n.Global.Info.success),\n",
+                "after": "param: callFunction(errorMessage ?? L10n.Global.Info.success),\n"
+            ]
+        ]
+    )
+
+    // MARK: SingleLineGuard
+    try Lint.checkFileContents(
+        checkInfo: "SingleLineGuard: Use a single line guard for simple checks.",
```

Comment:
Codacy found an issue: [Always start a multiline guard via `guard` then a line break and all expressions indented.](https://app.codacy.com/gh/Flinesoft/HandySwift/pullRequest?prid=5730214)